### ISSUE TITLE
"cannot unmarshal DNS message" when running "docker pull"

### ISSUE THREAD

**Issue by ztl8702** [docker/for-win#249]
### Expected behavior

Correctly pull Docker images.

### Actual behavior
`cannot unmarshal DNS message` when pulling image.

```bash
PS > docker pull redis
Using default tag: latest
Error response from daemon: Get https://registry-1.docker.io/v2/library/redis/manifests/latest: Get https://auth.docker.io/token?scope=repository%3Alibrary%2Fredis%3Apull&service=registry.docker.io: dial tcp: lookup auth.docker.io on 192.168.65.9:53: cannot unmarshal DNS message
```
Same error occurs when logging in.

```bash
PS > docker login
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username: <redacted>
Password:
Error response from daemon: Get https://registry-1.docker.io/v2/: Get https://auth.docker.io/token?account=<redacted>&client_id=docker&offline_token=true&service=registry.docker.io: dial tcp: lookup auth.docker.io on 192.168.65.9:53: cannot unmarshal DNS message
```

### Information

  - Diagnostic ID:  B1E27FF3-9EA0-4490-B603-1A6A378E0748/2016-11-23_15-59-47
  - Windows 10 Education Build 14965
  - Docker version 1.12.3, build 6b644ec
  - DNS server already set to "Fixed" (OpenDNS) in Docker settings:
![capture](https://cloud.githubusercontent.com/assets/2714446/20551617/5273a2a8-b196-11e6-946a-0736a8e13485.PNG)
     - changing DNS server to `8.8.8.8` results in the same behaviour.

### Steps to reproduce the behavior

  1. Download and install Docker for Windows.
  2. Set DNS to fixed: 208.67.222.222
  3. Run `docker pull redis` (or any other image names) in PowerShell.

### Notes

I tried to ssh into the Hyper-V virtual machine in order to change DNS inside it. But `docker-machine ls` shows that there is no hosts even though I can see a virtual machine in Hyper-V manager.

**Comment by ztl8702** [comment]
Log file attached.
[log.txt](https://github.com/docker/for-win/files/608131/log.txt)

**Comment by ztl8702** [comment]
Update:
Adding auth.docker.io to `hosts` (on Windows) seems to be a (partial) workaround.

```bash
Using default tag: latest
latest: Pulling from library/redis

386a066cd84a: Pulling fs layer
769149e3a45c: Pulling fs layer
1f43b3c0854a: Pulling fs layer
70e928127ad8: Waiting
9ad9c0058d76: Waiting
bc845722f255: Waiting
105d1e8cd76a: Waiting
error pulling image configuration: Get https://dseasb33srnrn.cloudfront.net/registry-v2/docker/registry/v2/blobs/sha256/5f/5f515359c7f871387c7dbb9bcf5c56b971d6b48ec9fec9e68d9369f4673db218/data?Expires=1479879842&Signature=aSDyTb8lbZA5u3RYfZ~DFPuE7g0pMkXmiYXuX2sORbvrkysRNIQTwTw5EoVVU~LrVZWFKH6IqcHbw7hf5rudPItsxU0TI2bW0RP0LY6Rn8NdA6hfiWRZ3NZXrebweV2eiugNmCYwBk1ggEPkiVmcojFmpDrNDq4GdTXajcm11p8_&Key-Pair-Id=APKAJECH5M7VWIS5YZ6Q: dial tcp: lookup dseasb33srnrn.cloudfront.net on 192.168.65.1:53: no such host
```

But it looks like I would have to add every domain name docker uses to `hosts`, which is impractical. So this is an indication that DNS is not working at all?

**Comment by rneugeba** [comment]
@ztl8702 thanks for your report. @djs55 does this error message rings a bell?

**Comment by djs55** [comment]
I've not seen that particular error before :( There have been a lot of DNS fixes in the beta channel however -- if possible could you try again with the latest beta 30.1?

**Comment by ztl8702** [comment]
Yeah actually I just found out that beta 30.1 works for me.

**Comment by rneugeba** [comment]
@ztl8702 glad to hear. Closing this issue. The fixes which are in Beta 30.1 will make it into the next stable release.
